textures/metro_control_panel_001_pegboard.py

```diff
@@ -25,10 +40,83 @@ def _remove_empty(pegs: Pegs) -> Pegs:
 
 
 def _swap_pegs(pegs: Pegs, a: str, b: str) -> Pegs:
+    """Trade places of two symbols"""
     assert len(a) == 1 and len(b) == 1
 
     placeholder = "_"
     # noinspection PyTypeChecker
     return tuple(
         tuple(s.replace(a, placeholder).replace(b, a).replace(placeholder, b) for s in panel) for panel in pegs
     )
+
+
+def matching_elements(sequence: Sequence[T], needle: T) -> Tuple[bool]:
+    """
+    Returns a tuple of bools equal length to sequence with True values
+    at every index where needle is equal to the item.
+
+    :param sequence: The sequence to check for matches.
+    :param needle: An object to check against the sequence.
+    :return: A tuple of boolean values corresponding if a match was found.
+
+    >>> matching_elements("PME", "M")
+    (False, True, False)
+    """
+    return tuple(item == needle for item in sequence)
+
+
+def invert_bools(bits: Iterable[bool]) -> Tuple[bool]:
+    return tuple(not bit for bit in bits)
+
+
+def bools_to_nor_mask(masking_bools: Collection[bool]) -> NorMask:
+    """
+    Creates an elementwise function that can called with two lists of booleans
+    of the same shape.
+    How the returned nor mask function is meant to be used:
+    (A, B, and C are lists; * is the nor mask function)
+    A * B = C
+    For C[n] the value A[n] is given where masking_bools[n] is False
+    and A[n] NOR B[n] is given where masking_bools[n] is True.
```

Comment:
I think something like that would be more clear (although it doesn't match the implementation):

```suggestion
    The function returns a sequence of binary functions. Depending on
    masking_bools[i], it returns either nor(a,b) if True, or left(a,b):=a otherwise.
    E.g., bools_to_nor_mask((True, False)) returns (nor, left) ...
```

Reply:
This was a great suggestion, I sometimes struggle to get the wording right.